crates/solver-service/src/apis/quote/mod.rs

```diff
@@ -112,7 +112,8 @@ pub async fn process_quote_request(
 
 	// Generate quotes using the business logic layer
 	let settlement_service = solver.settlement();
-	let quote_generator = QuoteGenerator::new(settlement_service.clone());
+	let delivery_service = solver.delivery();
+	let quote_generator = QuoteGenerator::new(settlement_service.clone(), delivery_service.clone());
```

Comment:
If we're not using delivery_service anywhere else, could we just use solver.delivery() directly in the argument? Then we don't need to clone?